src/main/java/ch/njol/skript/variables/FlatFileStorage.java

```diff
@@ -191,9 +198,62 @@ protected boolean load_i(final SectionNode n) {
 			Skript.info(file.getName() + " successfully updated.");
 		}
 		
+		if (SkriptConfig.getPreviousConfigVersion().isSmallerThan(v_2_4_beta9)) {
+			try {
+				Skript.info("Updating your config to version 2.4-beta9...");
+				r = new BufferedReader(new InputStreamReader(new FileInputStream(SkriptConfig.getConfig().getFile()), UTF_8));
+				List<String> convertedCfg = new ArrayList<>();
+				String line = null;
+				while((line = r.readLine()) != null) {
+					convertedCfg.add(line);
+					if (line.contains("type: CSV") || line.contains("type: SQLite")) {

```

Comment:
Bad check

Reply:
Wait, you are adding these options into the default storage setting section, but the saving code is global.
Don't forget that there can be multiple flat file storages (default is one of them with name 'default'), and a FlatFileStorage is equal to a single storage.

I mean, the saving code should be specific to a single storage (see how other storage specific options work), or at least the options should be in the global section.

Reply:
Does the current code not work with that? Isn't a SectionNode referring to a section of the config and not the whole thing?

Reply:
Just tested, this code supports multiple flat file databases, with each database having its own save interval and min. variable threshold. However, I do agree that the check has issues and needs to be reworked.